src/query-rule-custom-data/__tests__/query-rule-custom-data.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { createRenderer } from '../../../helpers/test-renderer';
+import { NgAisQueryRuleCustomData } from '../query-rule-custom-data';
+
+const render = createRenderer({
+  defaultState: {
+    items: [{ banner: 'lol yes it is a banner' }],
+  },
+  template: '<ais-query-rule-custom-data></ais-query-rule-custom-data>',
```

Comment:
We could add a test for the case using `context` to cover that use case too.

Reply:
I'm not following what you mean. Do you want a test with both widgets?

Reply:
I just mean write a test with the following template: 
```html
<ais-query-rule-custom-data>
   <ng-template>
    ...
   </ng-template>
</ais-query-rule-custom-data>
```
I think it would make sense since it's a different feature, and if a contributor breaks it inadvertently, a test would fire up an alarm.

Reply:
ah you mean "items"

Reply:
yes, or just ng-template 😄